test/solvers.jl

```diff
@@ -13,22 +13,17 @@ mos = try_import(:MosekTools)
 csd = try_import(:CSDP)
 scs = try_import(:SCS)
 
-# See https://github.com/JuliaOpt/JuMP.jl/pull/1396
-function with_opt(mod::Module, args...; kwargs...)
-    JuMP.OptimizerFactory(mod.Optimizer, args, kwargs)
-end
-
 # Semidefinite factories
-sdp_factories = JuMP.OptimizerFactory[]
+sdp_factories = MOI.OptimizerWithAttributes[]
 # Need at least Mosek 8 for sosdemo3 to pass; see:
 # https://github.com/JuliaOpt/Mosek.jl/issues/92
 # and at least Mosek 8.0.0.41 for sosdemo6 to pass; see:
 # https://github.com/JuliaOpt/Mosek.jl/issues/98
-mos && push!(sdp_factories, with_optimizer(MosekTools.Mosek.Optimizer, QUIET=true))
+mos && push!(sdp_factories, optimizer_with_attributes(MosekTools.Mosek.Optimizer, "QUIET" => true))
 # Currently, need to create a file param.csdp to avoid printing, see https://github.com/JuliaOpt/CSDP.jl/issues/15
-csd && push!(sdp_factories, with_opt(CSDP, printlevel=0))
-iscsdp(factory) = occursin("CSDP", string(factory.constructor))
+# csd && push!(sdp_factories, optimizer_with_attributes(CSDP.Optimizer, "printlevel" => 0))
```

Comment:
Why is this commented ?

Reply:
I should uncomment it. `Model(optimizer_with_attributes(CSDP.Optimizer, "printlevel" => 0.1))` would error. What's the best thing to do with solvers that haven't been updated for string parameters?

Reply:
Open an issue, I wasn't aware CSDP was not updated yet :)